src/RageTimer.cpp

```diff
@@ -141,18 +146,18 @@ RageTimer RageTimer::Sum(const RageTimer &lhs, float tm)
 	return ret;
 }
 
-float RageTimer::Difference(const RageTimer &lhs, const RageTimer &rhs)
+double RageTimer::Difference(const RageTimer& lhs, const RageTimer& rhs)
 {
-	int secs = lhs.m_secs - rhs.m_secs;
-	int us = lhs.m_us - rhs.m_us;
+	int64_t secs = lhs.m_secs - rhs.m_secs;
+	int64_t us = lhs.m_us - rhs.m_us;
 
-	if( us < 0 )
+	if (us < 0)
 	{
 		us += TIMESTAMP_RESOLUTION;
 		--secs;
 	}
 
-	return float(secs) + float(us) / TIMESTAMP_RESOLUTION;
+	return double(secs) + double(us) / TIMESTAMP_RESOLUTION;
```

Comment:
(style) Could these be static_cast please?

functional-style casts are legal c++, but a little odd to see - `double(x)` and `(double)x` are the same in this case, but same issue as c-casts on more complex types. static_cast, const_cast, dynamic_cast etc should all be preferred for modern c++ and general readability.